### PR TITLE
doc: improve documentation of BlockParams.MaxBytes

### PR DESCRIPTION
**Pull request by cason**
Additions to the description of the `BlockParams.MaxBytes` consensus parameter on the ABCI documentation.

Rendered: https://github.com/cometbft/cometbft/blob/cason/doc-maxbytes-param/spec/abci/abci%2B%2B_app_requirements.md

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [x] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

File: spec/abci/abci++_app_requirements.md [hunk 1]
```diff
@@ -601,6 +601,13 @@ This is enforced by the consensus algorithm.
 This implies a maximum transaction size that is this `MaxBytes`, less the expected size of
 the header, the validator set, and any included evidence in the block.
 
+The Application should be aware that honest validators _may_ produce and
+broadcast blocks with up the configured `MaxBytes` size.
+As a result, the consensus
+[timeout parameters](../../docs/core/configuration.md#consensus-timeouts-explained)
+adopted by nodes should be configured so that to account for the worst-case
```
Comment (sergio-mena) [inline spec/abci/abci++_app_requirements.md:608]:
```suggestion
adopted by nodes should be configured so as to account for the worst-case
```

File: spec/abci/abci++_app_requirements.md [hunk 2]
```diff
@@ -622,6 +622,13 @@ If the Application sets value -1, consensus will:
 
 Must have `MaxBytes == -1` OR `0 < MaxBytes <= 100 MB`.
 
+> Bear in mind that the default value for the `BlockParams.MaxBytes` consensus
+> parameter considers as valid blocks with size up to 21 MB.
```
Comment (sergio-mena) [inline spec/abci/abci++_app_requirements.md:626]:
```suggestion
> parameter accepts as valid blocks with size up to 21 MB.
```

File: spec/abci/abci++_app_requirements.md [hunk 3]
```diff
@@ -601,6 +601,13 @@ This is enforced by the consensus algorithm.
 This implies a maximum transaction size that is this `MaxBytes`, less the expected size of
 the header, the validator set, and any included evidence in the block.
 
+The Application should be aware that honest validators _may_ produce and
+broadcast blocks with up the configured `MaxBytes` size.
```
Comment (glnro) [inline spec/abci/abci++_app_requirements.md:605]:
```suggestion
broadcast blocks with up to the configured `MaxBytes` size.
```

**Review by glnro — APPROVED** [review]

File: spec/abci/abci++_app_requirements.md (hunk 3 above)
Comment (sergio-mena) [reply spec/abci/abci++_app_requirements.md:605]: I caught that in the backport PR but not here 🤦. Thanks @glnro

**Review by jmalicevic — APPROVED** [review]

File: spec/abci/abci++_app_requirements.md [hunk 4]
```diff
@@ -615,6 +622,13 @@ If the Application sets value -1, consensus will:
 
 Must have `MaxBytes == -1` OR `0 < MaxBytes <= 100 MB`.
```
Comment (jmalicevic) [inline spec/abci/abci++_app_requirements.md:623]: If I remember correctly this -1 was introduced by @sergio-mena to allow for arbitrary block size or is there some other reason?  I know that in 0.34.x it is not allowed.

Comment (cason) [reply spec/abci/abci++_app_requirements.md:623]: Yes, this is probably only on `main`. 34 and 37 not for sure, v0.38 I'll have to double check when backporting.

Comment (sergio-mena) [reply spec/abci/abci++_app_requirements.md:623]: IIRC, this is on `main`, `v0.38.x`, `v0.37x` (as it tackles a corner case on `PrepareProposal`), but I might be mistaken

Comment (cason) [reply spec/abci/abci++_app_requirements.md:623]: In v0.37.x it is not implemented at consensus level, from this extract:

https://github.com/cometbft/cometbft/blob/v0.37.x/consensus/state.go#L1937

Comment (cason) [reply spec/abci/abci++_app_requirements.md:623]: In v0.38.x it is.

Comment (sergio-mena) [reply spec/abci/abci++_app_requirements.md:623]: So my memories weren't accurate

**Pull request merged**